### PR TITLE
Add git ignore and pre commit

### PR DESCRIPTION
**Pull request by ArthurH91**
Adding a git ignore for Python and ROS2, and the pre commit.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Python ignores
+*.pyc
+*.pyo
+*.pyd
+__pycache__/
```
Comment (MaximilienNaveau) [inline .gitignore:5]:
```suggestion
*__pycache__*
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# ROS 2 build system ignores
+build/
+install/
+log/
+.colcon/
+
+# ROS 2 specific ignores
+ros2_ws/build/
+ros2_ws/install/
+ros2_ws/log/
```
Comment (MaximilienNaveau) [inline .gitignore:21]:
```suggestion
```

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# Ignore launch and configuration backup files
+*.launch.py~
+*.yaml~
+*.yml~
+*.xml~
```
Comment (MaximilienNaveau) [inline .gitignore:27]:
```suggestion
*~
```

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+*.iml
+*.DS_Store
+*.sublime-workspace
+*.sublime-project
+
+# CMake ignores
+CMakeCache.txt
+CMakeFiles/
+cmake_install.cmake
+Makefile
```
Comment (MaximilienNaveau) [inline .gitignore:41]: Only CMake build folder files not needed here.
```suggestion
```

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+# ROS-specific files
+*.bag
+*.db3  # For ROS 2 bag files
+*.rviz
+*.rqt_cache
+
+# Test-related ignores
+tests/__pycache__/
+tests/*.pyc
```
Comment (MaximilienNaveau) [inline .gitignore:51]: Already covered by the first lines
```suggestion
```

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    rev: v5.0.0
+    hooks:
+    -   id: check-added-large-files
+    -   id: check-ast
+    -   id: check-executables-have-shebangs
+    -   id: check-json
+    -   id: check-merge-conflict
+    -   id: check-symlinks
+    -   id: check-toml
+    # -   id: check-yaml
```
Comment (MaximilienNaveau) [inline .pre-commit-config.yaml:19]:
```suggestion
    -   id: check-yaml
```

**Review by MaximilienNaveau — CHANGES_REQUESTED** [review]
Take a look at the recommendations.

**Comment by ArthurH91** [comment]
Took your recommandations into account.

**Review by MaximilienNaveau — APPROVED** [review]

**Pull request merged**